### PR TITLE
[WIP] Change `groupby` default behavior to `sort=False`

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #5038

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/groupby/groupby.py [hunk 1]
```diff
@@ -30,9 +30,9 @@ class GroupBy(Serializable):
     level : int, level_name or list, optional
         For objects with a MultiIndex, `level` can be used to specify
         grouping by one or more levels of the MultiIndex.
-    sort : True, optional
-        If True (default), sort results by group9s). Note that
-        unlike Pandas, this also sorts values within each group.
+    sort : False, optional
+        If True, sort results by group's. Note that
+        unlike Pandas, this doesnot guarantee sorting values within each group.
```
Comment (kkraus14) [inline python/cudf/cudf/core/groupby/groupby.py:35]:
```suggestion
        If True, sort results by groups. Note that
        unlike Pandas, this does not guarantee sorting values within each group.
```

File: python/cudf/cudf/core/groupby/groupby.py [hunk 2]
```diff
@@ -82,6 +82,7 @@ def size(self):
             )
             .groupby(self.grouping)
             .agg("size")
+            .sort_index()
```
Comment (kkraus14) [inline python/cudf/cudf/core/groupby/groupby.py:85]: Why do we need to do a `sort_index` here?

File: python/cudf/cudf/tests/test_groupby.py [hunk 3]
```diff
@@ -1036,9 +1093,8 @@ def test_groupby_nth(n, by):
     )
     gdf = cudf.from_pandas(pdf)
 
-    expect = pdf.groupby(by).nth(n)
-    got = gdf.groupby(by).nth(n)
-
+    expect = pdf.groupby(by, sort=False).nth(n)
+    got = gdf.groupby(by, sort=False).nth(n)
```
Comment (kkraus14) [inline python/cudf/cudf/tests/test_groupby.py:1097]: Doesn't this default to `sort=False` and without sorting the order of output is arbitrary so we can't compare equality between cuDF / Pandas?

**Comment by kkraus14** [comment]
@galipremsagar lets retarget this to branch-0.15. This is too big to try to squeeze in for release.

File: python/cudf/cudf/core/groupby/groupby.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/groupby/groupby.py:85]: This is necessary for functions like `nth`.

File: python/cudf/cudf/tests/test_groupby.py (hunk 3 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_groupby.py]: @kkraus14 Correct, fixed the test to `sort=True` as we get arbitrary outputs.

But given that our sort=True is not consistent with pandas too, should we retain this test? I think yes, but mark it xfail if it fails in CI?

File: python/cudf/cudf/core/groupby/groupby.py [hunk 4]
```diff
@@ -185,7 +186,7 @@ def nth(self, n):
         Return the nth row from each group.
         """
         result = self.agg(lambda x: x.nth(n))
-        sizes = self.size()
+        sizes = self.size().sort_index()
```
Comment (kkraus14) [inline python/cudf/cudf/core/groupby/groupby.py:189]: I'm still not clear as to why we need to call `sort_index` here. `result` won't be sorted by the index above so the generated boolean mask wouldn't be valid here.

**Comment by kkraus14** [comment]
Given this has gone stale, closing this. We can reopen in the future.

**Pull request closed without merging**